plugins/modules/adguardhome_rewrite.py

```diff
@@ -0,0 +1,185 @@
+      - This assumes that the system controlled/queried by the module has not changed in a relevant way.
+
+
+author:
+  - Markus Bergholz (@markuman) <markuman+spambelongstogoogle@gmail.com>
+"""
+
+EXAMPLES = r"""
+- name: add dns rewrite rule in adguard home
+  community.dns.adguardhome_rewrite:
+    state: present
+    answer: 127.0.0.1
+    domain: example.org
+
+# when removing a rewrite, the current answer value must also match
+# therefore you can just leave it out and the existing value
+# will be used
+- name: remove rewrite for example.org
+  community.dns.adguardhome_rewrite:
+    state: absent
+    domain: example.org
+"""
+
+RETURN = r"""
+rules:
+  description: The list of fetched rewrite rules.
+  type: list
+  elements: dict
+  returned: always
+  contains:
+    answer:
+      description: Value of the rewrite.
+      type: str
+      sample: 192.168.178.71
+    domain:
+      description: Domain of the rewrite.
+      type: str
+      sample: dns.osuv.de
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.basic import missing_required_lib
+from ansible_collections.community.dns.plugins.module_utils.adguardhome.api import (
+    AdGuardHomeAPIHandler
+)
+import traceback
+
+YAML_IMP_ERR = None
+try:
+    import yaml
+    HAS_YAML = True
+except Exception:
+    YAML_IMP_ERR = traceback.format_exc()
+    HAS_YAML = False
+
+
+def find_and_compare(rules, domain, answer):
+    domain_exists = False
+    value_is_different = False
+    target = {}
+    for rule in rules:
+        if rule["domain"] == domain:
+            domain_exists = True
+            target = {"domain": domain, "answer": rule["answer"]}
+            if rule["answer"] != answer:
+                value_is_different = True
+            break
+    return domain_exists, value_is_different, target
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            username=dict(type='str', required=True),
+            password=dict(type='str', required=True, no_log=True),
+            host=dict(type='str', required=True),
+            ssl_verify=dict(type='bool', default=True),
+            state=dict(type='str', default='present', choices=['present', 'absent']),
+            answer=dict(type='str', required=False),
+            domain=dict(type='str', required=True),
+        ),
+        supports_check_mode=True,
+        required_if=[['state', 'present', ['answer']]],
+    )
+
+    if not HAS_YAML:
+        module.fail_json(
+            msg=missing_required_lib("pyyaml", url='https://pyyaml.org/wiki/PyYAMLDocumentation'),
+            exception=YAML_IMP_ERR
+        )
+
+    domain = module.params.get('domain')
+    answer = module.params.get('answer')
+    state = module.params.get('state')
+
+    adguardhome = AdGuardHomeAPIHandler(module.params, module.fail_json)
+
+    before = adguardhome.list()
+    changed = False
+
+    DOMAIN_EXISTS, VALUE_IS_DIFFERENT, TARGET = find_and_compare(before, domain, answer)
```

Comment:
Upper-case identifiers are reserved for global constants. Please use lower-case letters.